modules/porous_flow/src/materials/PorousFlowPorosityConst.C

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowPorosityConst.h"
+
+#include "Conversion.h"
+
+template<>
+InputParameters validParams<PorousFlowPorosityConst>()
+{
+  InputParameters params = validParams<PorousFlowPorosityUnity>();
+  params.addRequiredParam<Real>("porosity", "The porosity, which is assumed constant for this material");
+  params.addClassDescription("This Material calculates the porosity assuming it is constant");
+  return params;
+}
+
+PorousFlowPorosityConst::PorousFlowPorosityConst(const InputParameters & parameters) :
+    PorousFlowPorosityUnity(parameters),
+    _input_porosity(getParam<Real>("porosity"))
+{
+}
+
+void
+PorousFlowPorosityConst::initQpStatefulProperties()
+{
+  _porosity_nodal[_qp] = _input_porosity; // this becomes _porosity_old[_qp] in the first call to computeQpProperties
+  _porosity_qp[_qp] = _input_porosity; // this becomes _porosity_old[_qp] in the first call to computeQpProperties
+
+  const unsigned int num_var = _dictator_UO.numVariables();
+  _dporosity_nodal_dvar[_qp].resize(num_var, 0.0);
+  _dporosity_qp_dvar[_qp].resize(num_var, 0.0);
+  _dporosity_nodal_dgradvar[_qp].resize(num_var, RealGradient());
+  _dporosity_qp_dgradvar[_qp].resize(num_var, RealGradient());
```

Comment:
using `assign` rather than `resize` would be a bit more expressive here (resize will only initialize the newly allocated entries, whereas assign makes it clear that you want the entire vector to be initialized with the second argument)